tests/functional/algos/test_algos.py

```diff
@@ -247,3 +247,64 @@ def test_with_evc(algorithm):
         param = best_trial._params[1]
         assert param.name == "x"
         assert param.type == "real"
+
+
+@pytest.mark.parametrize(
+    "algorithm", algorithm_configs.values(), ids=list(algorithm_configs.keys())
+)
+def test_parallel_workers(algorithm, tmp_path):
+    """Test parallel execution with joblib"""
+    pickle_host = str(tmp_path / "db.pkl")
```

Comment:
It would be better to use OrionState to make sure we handle cleanup properly. Using `with OrionState(options...) as storage_state:` you can access storage config with `storage_state.storage_config` and pass it to `create_experiment`.

Reply:
I tried this, but it did not work when put the code into `with OrionState`. After fixed the evc issue this time, still fail when putting code into `with OrionState(storage=storage):`, only the first random algo succeeds, all the other algorithms will fail because some un-consistent storage issue in `OrionState context` and in the `_optimize method` although they have exactly the same storage configure.

Reply:
It will good if you can also give it a try. Or maybe we can take a look of this potential issue later.

Reply:
It will be good if you can also give it a look/try of this behavior Or maybe we can take a look of this potential issue later? Seems it should not be a functional issue.

Reply:
Hmm, I would be concerned that users stumble on a similar issue. I would be in favor of investigating the issue and fixing it inside this PR.

Reply:
I think this is the only issue left. Would you like me to take a look?

Reply:
It will be helpful if you can take a look. thanks

Reply:
I found the issue. The problem is that storage_config is None if no custom configuration for the storage is passed. The worker will end up creating a storage based on global configuration, but the local worker may already have a storage initiated with a custom configuration, only this configuration was not passed to `build_experiment`. I think it may be easier to get around this issue by making the DB picklable instead of changing the experiment. I will give it a try.

Reply:
I made a PR on your branch with the fix: https://github.com/donglinjy/orion/pull/3